### PR TITLE
Hide command line merge instructions when user can't push

### LINKED ISSUE #17121: Showing command line instructions on pull request despite branch protection existing
### Gitea Version

1.15.3

### Git Version

N/A

### Operating System

N/A

### How are you running Gitea?

Running on a Linux machine

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Description

https://try.gitea.io/my1e5_test_user/Test-Repo/pulls/1

I have branch protection enabled on my 'master' branch so that "No pushing will be allowed to this branch"

image

However, on my pull request it shows these command line instructions which technically are not valid as I have branch protection enabled. `git push origin master` wouldn't work.

image

I feel this is slightly inelegant. Is there a way to make it so that if branch protection is enabled, these command line instructions are not shown?

Or alternatively, create a setting that allows you to disable showing these command line instructions? I don't want it to confuse users - I've had a user ask why these instructions are showing when branch protection is enabled.

Thanks!

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by qwerty287**
Hides the command line instructions if the authenticated user can't push to the base branch. 
Closes #17121

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: routers/web/repo/issue.go [hunk 1]
```diff
@@ -1543,6 +1543,7 @@ func ViewIssue(ctx *context.Context) {
 			ctx.ServerError("LoadProtectedBranch", err)
 			return
 		}
+		ctx.Data["ShowMergeInstructions"] = true
```
Comment (delvh) [inline routers/web/repo/issue.go:1546]: I think instead of `true` by default, it would be better to use whether the user has write rights for this repo.

Comment (delvh) [reply routers/web/repo/issue.go:1546]: Or is that intended for local merging? In that case, `true` is the correct value.

Comment (qwerty287) [reply routers/web/repo/issue.go:1546]: If you have write permissions that doesn't not mean that you can push to the branch

Comment (delvh) [reply routers/web/repo/issue.go:1546]: Yes, if there is branch protection, which is the line you added below.
But if you cannot even write to the repo, you most likely won't be able to write to the protected branch as well.

Comment (qwerty287) [reply routers/web/repo/issue.go:1546]: If you can't merge the PR these instructions won't be shown (and you can't merge without write permission)

Comment (delvh) [reply routers/web/repo/issue.go:1546]: Ah, I see. Then the current `true` suffices.

**Review by delvh — APPROVED** [review]

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**